cmd/coder/main.go

```diff
@@ -15,6 +20,48 @@ import (
 func main() {
 	rand.Seed(time.Now().UnixMicro())
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Add a custom SIGQUIT handler that outputs to stderr and a well-known file
+	// in the home directory. This also prevents SIGQUITs from killing the CLI.
+	go func() {
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGQUIT)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-sigs:
+			}
+
+			buf := make([]byte, 10_000_000)
```

Comment:
The Go runtime package has a nice method for this to avoid the up-front allocation: https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/runtime/debug/stack.go;l=22-29. It doesn't have an upper limit, so we may want to inline it here with a maximum size.

Reply:
Inlined it and added a check for 64MB max buffer size, starting at 1MB. 1KB isn't a big enough starting size because the server and agent make many goroutines.